scripts/linux/electron-installer-appimage.sh

```diff
@@ -102,17 +101,14 @@ else
   exit 1
 fi
 
-if [ -z "$TMPDIR" ]; then
-  TMPDIR=$(mktemp -d)
-fi
-mkdir -p "$TMPDIR"
+APPIMAGE_TMPDIR=$(mktemp --dry-run -d -t "$ARGV_APPLICATION_NAME-linux-$ARGV_ARCHITECTURE-XXXX")
```

Comment:
Yeah, the thing is that this file downloads and creates various other files during its life, which are hard to declare in the Makefile (since they happen within the script). You can think of this script as having an "resource app asar input", and an "appimage output". In order to perform that task, a couple of tools have to be downloaded, and an AppDir needs to be created.

How do you think we can handle that while still being make-friendly?

Here's a suggestion: we have 4 external artifacts during this script:

- `AppRun`
- `desktopintegration`
- `AppImageAssistant`
- The `AppDir`

The `AppRun` and `desktopintegration` files are downloaded straight to the AppDir, and the `AppImageAssistant` is only required to convert the AppDir to an AppImage.

So we can do the following things:

- Create a new script whose mission is to just create an AppDir. This script will download `AppRun` and `desktopintegration` straight to the output directory (the AppDir), so no temporary stuff here.

- We create another script to convert an AppDir to an AppImage. By dividing things this way, the AppDir can now be declared in the Makefile, so thats something less to think about.

The only concern now is `AppImageAssistant`, which needs to be used by the script that goes from AppDir to AppImage. We can always create another script just to download it (and thus declare it in the Makefile), however that means that the Makefile will be responsible to know how to use it correctly, and I feel the Makefile shouldn't care about that.

Another alternative is to be able to pass the path to `AppImageAssistant` to the AppDir->AppImage script, however that feels weird as well.

What do you think?